analyzers/its/expected/Ember-MM/EmberAPI-{208AA35E-C6AE-4D2D-A9DD-B6EFD19A4279}-S2259.json

```diff
@@ -0,0 +1,82 @@
+{
+"issues":  [
+{
+"id":  "S2259",
+"message":  "'movieName' is Nothing on at least one execution path.",
+"location":  {
+"uri":  "sources\Ember-MM\EmberAPI\clsAPIStringUtils.vb",
+"region":  {
+"startLine":  161,
+"startColumn":  55,
+"endLine":  161,
+"endColumn":  64
+}
+}
+},
+{
+"id":  "S2259",
+"message":  "'movieName' is Nothing on at least one execution path.",
+"location":  {
+"uri":  "sources\Ember-MM\EmberAPI\clsAPIStringUtils.vb",
+"region":  {
+"startLine":  162,
+"startColumn":  60,
+"endLine":  162,
+"endColumn":  69
+}
+}
+},
+{
+"id":  "S2259",
+"message":  "'movieName' is Nothing on at least one execution path.",
+"location":  {
+"uri":  "sources\Ember-MM\EmberAPI\clsAPIStringUtils.vb",
+"region":  {
+"startLine":  164,
```

Comment:
These is a FPs, right? Let's add minimal C# UT to reproduce it

Reply:
I think so. I added a test case. I think the root cause is the assumption, that "IsNullOrEmpty" might throw, but I'm not sure. The graph for the test looks like that:
https://edotor.net/?engine=dot#%20%20%20%20digraph%20%22RoslynCfg%22%20%7B%0A%20%20%20%20subgraph%20%22cluster_1%22%20%7B%0A%20%20%20%20label%20%3D%20%22TryAndFinally%20region%22%0A%20%20%20%20subgraph%20%22cluster_2%22%20%7B%0A%20%20%20%20label%20%3D%20%22Try%20region%22%0A%20%20%20%20cfg0_block1%20%5Bshape%3Drecord%20label%3D%22%7BBLOCK%20%231%7C%23%23%20BranchValue%20%23%23%7C0%23%20InvocationOperation%3A%20IsNullOrEmpty%20%2F%20InvocationExpressionSyntax%3A%20string.IsNullOrEmpty(arg)%7C1%23%20ArgumentOperation%20%2F%20ArgumentSyntax%3A%20arg%7C2%23%20ParameterReferenceOperation%20%2F%20IdentifierNameSyntax%3A%20arg%7C%23%23%23%23%23%23%23%23%23%23%7D%22%5D%0A%20%20%20%20%7D%0A%20%20%20%20subgraph%20%22cluster_3%22%20%7B%0A%20%20%20%20label%20%3D%20%22Finally%20region%22%0A%20%20%20%20cfg0_block2%20%5Bshape%3Drecord%20label%3D%22%7BBLOCK%20%232%7C0%23%20ExpressionStatementOperation%20%2F%20ExpressionStatementSyntax%3A%20Tag(%5C%22ArgInFinally%5C%22%2C%20arg)%3B%7C1%23%20InvocationOperation%3A%20Tag%20%2F%20InvocationExpressionSyntax%3A%20Tag(%5C%22ArgInFinally%5C%22%2C%20arg)%7C2%23%20InstanceReferenceOperation%20%2F%20IdentifierNameSyntax%3A%20Tag%7C2%23%20ArgumentOperation%20%2F%20ArgumentSyntax%3A%20%5C%22ArgInFinally%5C%22%7C3%23%20LiteralOperation%20%2F%20LiteralExpressionSyntax%3A%20%5C%22ArgInFinally%5C%22%7C2%23%20ArgumentOperation%20%2F%20ArgumentSyntax%3A%20arg%7C3%23%20ConversionOperation%20%2F%20IdentifierNameSyntax%3A%20arg%7C4%23%20ParameterReferenceOperation%20%2F%20IdentifierNameSyntax%3A%20arg%7C%23%23%23%23%23%23%23%23%23%23%7D%22%5D%0A%20%20%20%20%7D%0A%20%20%20%20%7D%0A%20%20%20%20cfg0_block0%20%5Bshape%3Drecord%20label%3D%22%7BENTRY%20%230%7D%22%5D%0A%20%20%20%20cfg0_block3%20%5Bshape%3Drecord%20label%3D%22%7BEXIT%20%233%7D%22%5D%0A%20%20%20%20cfg0_block0%20-%3E%20cfg0_block1%0A%20%20%20%20cfg0_block2%20-%3E%20NoDestination_cfg0_block2%20%5Blabel%3D%22StructuredExceptionHandling%22%5D%0A%20%20%20%20cfg0_block1%20-%3E%20cfg0_block3%20%5Blabel%3D%22WhenFalse%22%5D%0A%20%20%20%20cfg0_block1%20-%3E%20cfg0_block3%20%5Blabel%3D%22Else%22%5D%0A%20%20%20%20%7D%0A